core/src/scene/drawRule.cpp

```diff
@@ -211,15 +210,17 @@ void DrawRuleMergeSet::apply(const Feature& _feature, const SceneLayer& _layer,
 
 void DrawRuleMergeSet::mergeRules(const SceneLayer& _layer) {
 
-    for (const auto& rule : _layer.rules()) {
+    size_t pos, end = m_matchedRules.size();
 
-        auto it = std::find_if(m_matchedRules.begin(), m_matchedRules.end(),
```

Comment:
No, one needs to only compare the id's up to ```m_matchedRules.size()``` from the beginning, since the rules from _layer don't have the same id's twice.